x-pack/plugins/security_solution/common/detection_engine/schemas/common/schemas.ts

```diff
@@ -459,12 +459,21 @@ export type Threats = t.TypeOf<typeof threats>;
 export const threatsOrUndefined = t.union([threats, t.undefined]);
 export type ThreatsOrUndefined = t.TypeOf<typeof threatsOrUndefined>;
 
-export const threshold = t.exact(
-  t.type({
-    field: t.string,
-    value: PositiveIntegerGreaterThanZero,
-  })
-);
+export const threshold = t.intersection([
+  t.exact(
+    t.type({
+      field: t.union([t.string, t.array(t.string)]),
+      value: PositiveIntegerGreaterThanZero,
+    })
+  ),
+  t.exact(
+    t.partial({
+      cardinality_field: t.union([t.string, t.array(t.string), t.undefined]),
+      cardinality_value: t.union([PositiveInteger, t.undefined]), // TODO: cardinality_value should be set if cardinality_field is set
```

Comment:
is there a reason to allow `cardinality_value` to be 0 here? The `PositiveInteger` type is named incorrectly and allows the input to be 0

Reply:
@marshallmain I think `0` is probably a valid case. You might want to check if a field is set at all (not null or undefined)...

Reply:
@marshallmain Hmm good point... the operator is supposed to be configurable (to either `<`, `<=`, `==`, `>`, `>=`), but that hasn't been implemented yet. I think I can address this separately in a follow-up PR unless you feel strongly about it.